lib/colorado_lottery.rb

```diff
@@ -0,0 +1,39 @@
+class ColoradoLottery
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants
+  def initialize(registered_contestants = {},
+                winners = [],
+                current_contestants = {})
+    @registered_contestants = registered_contestants
+    @winners = winners
+    @current_contestants = current_contestants
+  end
+
+  def interested_and_18?(contestant, game)
+    interested = contestant.game_interests.include?(game.name)
+    contestant.age >= 18 && interested
+  end
+
+  def can_register?(contestant, game)
+    return false unless interested_and_18?(contestant,game)
+    !contestant.out_of_state? || game.national_drawing?
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game)
```

Comment:
you don't need to check for this-- the interaction pattern says we will only call this method for contestants that can be successfully registered. but good thought.